test/jdk/javax/swing/JRadioButton/8033699/bug8033699.java

```diff
@@ -57,64 +57,80 @@ public class bug8033699 {
-        runTest3();
-        robot.delay(100);
-
-        // left/up key test in grouped radio button
-        runTest4();
-        robot.delay(100);
 
-        // down/right key test in grouped radio button
-        runTest5();
-        robot.delay(100);
-
-        // tab from radio button in group to next component in the middle of button group layout
-        runTest6();
-        robot.delay(100);
-
-        // tab to radio button in group from component in the middle of button group layout
-        runTest7();
-        robot.delay(100);
-
-        // down key circle back to first button in grouped radio button
-        runTest8();
-        robot.delay(100);
-
-        // Verify that ActionListener is called when a RadioButton is selected using arrow key.
-        runTest9();
-        robot.delay(100);
+        // Get all installed Look and Feels
+        UIManager.LookAndFeelInfo[] lafs = UIManager.getInstalledLookAndFeels();
+        for (UIManager.LookAndFeelInfo laf : lafs) {
+            testLaF(laf);
+        }
+    }
 
-        SwingUtilities.invokeAndWait(() -> mainFrame.dispose());
+    private static void testLaF(UIManager.LookAndFeelInfo laf) throws Exception {
+        try {
+            System.out.println("Testing LaF: " + laf.getName());
+            SwingUtilities.invokeAndWait(() -> {
+                setLookAndFeel(laf);
+                createAndShowGUI();
+            });
+
+            robot.waitForIdle();
+            robot.delay(1000);
+
+            // tab key test grouped radio button
+            runTest1();
+            robot.delay(100);
+
+            // tab key test non-grouped radio button
+            runTest2();
+            robot.delay(100);
+
+            // shift tab key test grouped and non-grouped radio button
+            runTest3();
+            robot.delay(100);
+
+            // left/up key test in grouped radio button
+            runTest4();
+            robot.delay(100);
+
+            // down/right key test in grouped radio button
+            runTest5();
+            robot.delay(100);
+
+            // tab from radio button in group to next component in the middle of button group layout
+            runTest6();
+            robot.delay(100);
+
+            // tab to radio button in group from component in the middle of button group layout
+            runTest7();
+            robot.delay(100);
+
+            // down key circle back to first button in grouped radio button
+            runTest8();
+            robot.delay(100);
+
+            // Verify that ActionListener is called when a RadioButton is selected using arrow key.
+            runTest9();
+            robot.delay(100);
+        } catch (Exception e) {
+            throw new RuntimeException("Error testing LaF: " + laf.getName(), e);
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                if (mainFrame != null) {
+                    mainFrame.dispose();
+                    mainFrame = null;
```

Comment:
Assigning `mainFrame = null` is not required since we are disposing the frame at this point.

Reply:
> Assigning `mainFrame = null` is not required since we are disposing the frame at this point.

It's required here, because the loop is repeated, and setting `mainFrame` to `null` prevents access to a disposed frame… likely from disposing of the frame again.

Reply:
Missed seeing the finally block is within the LaF loop. Looks fine then.